File: template/nestJs/src/auth/dto/create-api-token.dto.ts
import { ApiProperty } from '@nestjs/swagger';
import { IsEmail, IsString, IsOptional } from 'class-validator';

export class CreateApiTokenDto {
  @ApiProperty({
    description: '邮箱'
  })
  @IsEmail()
  email: string;

  @ApiProperty({
    description: '密码'
  })
  @IsString()
  password: string;

  @ApiProperty({
    description: '令牌名称'
  })
  @IsOptional()
  @IsString()
  tokenName?: string; // 可选的token名称，用于标识
}

File: template/springboot/src/main/java/com/TinyPro/entity/dto/CreateApiTokenDto.java
package com.TinyPro.entity.dto;

import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotEmpty;
import lombok.Data;

/**
 * API Token creation request, compatible with the NestJS backend.
 */
@Data
public class CreateApiTokenDto {

    @NotEmpty(message = "{NOT_EMPTY}")
    @Email
    private String email;

    @NotEmpty(message = "{NOT_EMPTY}")
    private String password;

    /** Optional token name. A random ID is generated when omitted. */
    private String tokenName;
}
